src/lib.rs

```diff
@@ -151,10 +164,28 @@ impl ProcessManager {
 
 	async fn restart_process(&self, process_key: ProcessKey) -> Result<Child> {
 		let mut inner = self.inner.write().await;
+		let restart_mode = inner.restart_mode;
 		let process_data = inner
 			.processes
 			.get_mut(process_key)
 			.ok_or(Error::InvalidProcess(process_key))?;
+
+		// delay before restarting
+		match restart_mode {
+			RestartMode::ExponentialBackoff(backoff) => {
```

Comment:
I've been thinking of integrating the https://crates.io/crates/backoff crate, which I guess would do the same thing but provide a few more options.